### PR TITLE
sql-statements: use EBNF to render syntax diagrams - third batch

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

This PR changes the synopsis sections of BEGIN, CHANGE, COMMIT, CREATE, DEALLOCATE, DELETE, DO, and DROP statements to use the ```ebnf+diagrm block, replacing the original image-based sqlgrams. It is a follow-up PR to https://github.com/pingcap/docs/issues/4389.

To avoid making one single PR too large, I will open 4-5 follow-up PRs to process sql-statements files in batches. And I will delete all image files in the last PR in case of any display issues.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://pingcap.github.io/sqlgram

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-explain.md [hunk 1]
```diff
@@ -26,6 +26,24 @@ TiDB 支持 `EXPLAIN [options] FOR CONNECTION connection_id`，但与 MySQL 的
 
 ![ExplainableStmt](/media/sqlgram/ExplainableStmt.png)
```
Comment (kennytm) [inline sql-statements/sql-statement-explain.md:27]: the images are still here 🤔

File: sql-statements/sql-statement-modify-column.md [hunk 2]
```diff
@@ -30,6 +30,42 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-modify-column/','/docs-cn/d
 
 ![ColumnPosition](/media/sqlgram/ColumnPosition.png)
```
Comment (kennytm) [inline sql-statements/sql-statement-modify-column.md:31]: ditto

File: sql-statements/sql-statement-replace.md [hunk 3]
```diff
@@ -34,6 +34,31 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-replace/','/docs-cn/dev/ref
 
 ![InsertValues](/media/sqlgram/InsertValues.png)
```
Comment (kennytm) [inline sql-statements/sql-statement-replace.md:35]: ditto

**Review by kennytm — COMMENTED** [review]
Rest LGTM

File: sql-statements/sql-statement-explain.md (hunk 1 above)
Comment (CharLotteiu) [reply sql-statements/sql-statement-explain.md]: Updated in https://github.com/pingcap/docs-cn/pull/5540/commits/4119ba2d72b724adf520f74aa6ef01c014eb8da2.

**Pull request merged**